--- /etc/inc/pkg-utils.inc.orig	2021-05-29 05:04:18.000000000 +0000
+++ /etc/inc/pkg-utils.inc	2021-06-08 11:05:29.153780000 +0000
@@ -403,7 +403,7 @@
 	if ($base_packages) {
 		$repo_param = "";
 	} else {
-		$repo_param = "-r {$g['product_name']}";
+               $repo_param = "";
 	}
 
 	/*
@@ -475,10 +475,15 @@
 	}
 
 	$result = array();
+	$ar_seen = array();
 	foreach ($search_items as $pkg_info) {
 		if (empty($pkg_info['name'])) {
 			continue;
-		}
+		}	
+		// remove duplicates specifically for WG package
+		if ( $ar_seen[$pkg_info['name']] == 1) 
+			continue;
+		$ar_seen[$pkg_info['name']] = 1;
 
 		if (isset($pkg_filter) && !in_array($pkg_info['name'],
 		    $pkg_filter)) {
@@ -500,7 +505,7 @@
 			    $err);
 			if (!$base_packages &&
 			    rtrim($out) != $g['product_name']) {
-				continue;
+                               //continue;
 			}
 
 			$pkg_info['installed'] = true;
